client/core/core.go

```diff
@@ -1595,29 +1614,21 @@ func (c *Core) ReconfigureWallet(appPW []byte, assetID uint32, cfg map[string]st
 	return nil
 }
 
-// SetWalletPassword updates the (encrypted) password for the wallet.
-func (c *Core) SetWalletPassword(appPW []byte, assetID uint32, newPW []byte) error {
+// setWalletPassword updates the (encrypted) password for the wallet.
+func (c *Core) setWalletPassword(appPW []byte, wallet *xcWallet, newPW []byte) error {
 	// Check the app password and get the crypter.
 	crypter, err := c.encryptionKey(appPW)
 	if err != nil {
-		return newError(authErr, "SetWalletPassword password error: %v", err)
+		return newError(authErr, "setWalletPassword password error: %v", err)
 	}
 
 	newPasswordSet := len(newPW) > 0
 
-	// Check that the specified wallet exists.
-	c.walletMtx.Lock()
-	defer c.walletMtx.Unlock()
-	wallet, found := c.wallets[assetID]
-	if !found {
-		return newError(missingWalletErr, "wallet for %s (%d) is not known", unbip(assetID), assetID)
-	}
-
-	// Connect if necessary.
 	wasConnected := wallet.connected()
 	if !wasConnected {
 		if err = c.connectWallet(wallet); err != nil {
-			return newError(connectionErr, "SetWalletPassword connection error: %v", err)
+			return newError(connectionErr, "setWalletPassword connection error: %v",
+				err)
 		}
 	}
 
```

Comment:
I noticed we have an existing issue with the password check below.  That is, it only checks the password if one is set, but it might be incorrect to not have one set. 

In the `// Check that the new password works...` comment below, maybe create a TODO:

```go
// TODO: find a way to verify that the wallet actually is unencrypted or otherwise does
// not require a password. Perhaps an asset.Wallet.RequiresPassword wallet method?
```

I'm thinking of a method added to client/asset.Wallet like:

```go
// RequiresPassword indicates if the wallet requires a password for certain
// operations such as sending funds.
// RequiresPassword(bool, error) // TODO
```